homeassistant/components/xiaomi_miio/humidifier.py

```diff
@@ -0,0 +1,382 @@
+"""Support for Xiaomi Mi Air Purifier and Xiaomi Mi Air Humidifier with humidifier entity."""
+from enum import Enum
+import logging
+import math
+
+from miio.airhumidifier import OperationMode as AirhumidifierOperationMode
+from miio.airhumidifier_miot import OperationMode as AirhumidifierMiotOperationMode
+
+from homeassistant.components.humidifier import HumidifierEntity
+from homeassistant.components.humidifier.const import (
+    DEFAULT_MAX_HUMIDITY,
+    DEFAULT_MIN_HUMIDITY,
+    DEVICE_CLASS_HUMIDIFIER,
+    SUPPORT_MODES,
+)
+from homeassistant.const import ATTR_MODE, CONF_HOST, CONF_TOKEN
+from homeassistant.core import callback
+from homeassistant.util.percentage import percentage_to_ranged_value
+
+from .const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    DOMAIN,
+    KEY_COORDINATOR,
+    KEY_DEVICE,
+    MODEL_AIRHUMIDIFIER_CA1,
+    MODEL_AIRHUMIDIFIER_CA4,
+    MODEL_AIRHUMIDIFIER_CB1,
+    MODELS_HUMIDIFIER_MIOT,
+    SERVICE_SET_TARGET_HUMIDITY,
+)
+from .device import XiaomiMiioEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Xiaomi Miio Device"
+DATA_KEY = "fan.xiaomi_miio"
+
+CONF_MODEL = "model"
+
+ATTR_MODEL = "model"
+
+# Air Humidifier
+ATTR_TARGET_HUMIDITY = "target_humidity"
+ATTR_TRANS_LEVEL = "trans_level"
+ATTR_HARDWARE_VERSION = "hardware_version"
+
+SERVICE_TO_METHOD = {
+    SERVICE_SET_TARGET_HUMIDITY: {
+        "method": "async_set_target_humidity",
+    },
+}
+
+AVAILABLE_ATTRIBUTES = {
+    ATTR_MODE: "mode",
+    ATTR_TARGET_HUMIDITY: "target_humidity",
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Fan from a config entry."""
+    entities = []
+
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
```

Comment:
Invert and return early. And do this check first before anything else.

Reply:
Done